java-compiler-testing/src/main/java/io/github/ascopes/jct/compilers/impl/JctCompilationFactory.java

```diff
@@ -59,39 +66,48 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(JctCompilationFactory.class);
 
+  private final Workspace workspace;
+  private final A compiler;
+  private final JavaCompiler jsr199Compiler;
+  private final JctFlagBuilder flagBuilder;
+
+  private final Lazy<List<Path>> jvmClassPath;
+  private final Lazy<List<Path>> jvmModulePath;
+  private final Lazy<List<Path>> jvmPlatformPath;
+  private final Lazy<List<Path>> jvmSystemModules;
+
   /**
    * Initialise this factory.
    */
-  public JctCompilationFactory() {
-    // Nothing to do here.
-  }
+  public JctCompilationFactory(
+      Workspace workspace,
+      A compiler,
+      JavaCompiler jsr199Compiler,
+      JctFlagBuilder flagBuilder
+  ) {
+    this.workspace = workspace;
+    this.compiler = compiler;
+    this.jsr199Compiler = jsr199Compiler;
+    this.flagBuilder = flagBuilder;
+    jvmClassPath = new Lazy<>(SpecialLocationUtils::currentClassPathLocations);
+    jvmModulePath = new Lazy<>(SpecialLocationUtils::currentModulePathLocations);
+    jvmPlatformPath = new Lazy<>(SpecialLocationUtils::currentPlatformClassPathLocations);
+    jvmSystemModules = new Lazy<>(SpecialLocationUtils::javaRuntimeLocations);
 
-  @Override
-  public String toString() {
-    return new ToStringBuilder(this).toString();
   }
 
   /**
    * Run the compilation for the given compiler and return the compilation result.
    *
-   * @param compiler       the compiler to run.
-   * @param builder        the template to compile files within.
-   * @param jsr199Compiler the underlying JSR-199 compiler to run via.
-   * @param flagBuilder    the flag builder to use.
    * @return the compilation result.
    */
-  public JctCompilationImpl compile(
-      A compiler,
-      JctFileManagerBuilder builder,
-      JavaCompiler jsr199Compiler,
-      JctFlagBuilder flagBuilder
-  ) {
+  public JctCompilationImpl build() {
     try {
       var flags = buildFlags(compiler, flagBuilder);
       var diagnosticListener = buildDiagnosticListener(compiler);
       var writer = buildWriter(compiler);
 
-      try (var fileManager = buildFileManager(compiler, builder)) {
+      try (var fileManager = buildFileManager()) {
```

Comment:
- [ ] Remove closable semantics for JctFileManager as it gets closed by the compiler and this does not make sense when we perform one or more compiler passes. Nothing is closed now either.